src/VirtualClient/VirtualClient.Monitors/AmdSmiMonitor.cs

```diff
@@ -0,0 +1,132 @@
+                    break;
+            }
+        }
+
+        /// <inheritdoc/>
+        protected void ValidateParameters()
+        {
+            if (this.MonitorFrequency <= TimeSpan.Zero)
+            {
+                throw new MonitorException(
+                    $"The monitor frequency defined/provided for the '{this.TypeName}' component '{this.MonitorFrequency}' is not valid. " +
+                    $"The frequency must be greater than zero.",
+                    ErrorReason.InvalidProfileDefinition);
+            }
+        }
+
+        /// <summary>
+        /// Query the gpu for utilization information
+        /// </summary>
+        /// <param name="telemetryContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        private async Task QueryGpuAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            ISystemManagement systemManagement = this.Dependencies.GetService<ISystemManagement>();
+            IFileSystem fileSystem = systemManagement.FileSystem;
+
+            int totalSamples = (int)this.MonitorFrequency.TotalSeconds;
+            string command = "amdsmi";
+            string commandArguments = "metric --csv";
+
+            await Task.Delay(this.MonitorWarmupPeriod, cancellationToken)
+                .ConfigureAwait(false);
+
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                try
+                {
+                    using (IProcessProxy process = systemManagement.ProcessManager.CreateElevatedProcess(this.Platform, command, $"{commandArguments}", Environment.CurrentDirectory))
+                    {
+                        this.CleanupTasks.Add(() => process.SafeKill());
+
+                        DateTime startTime = DateTime.UtcNow;
+                        await process.StartAndWaitAsync(cancellationToken)
+                            .ConfigureAwait(false);
+
+                        DateTime endTime = DateTime.UtcNow;
+
+                        if (!cancellationToken.IsCancellationRequested)
+                        {
+                            try
+                            {
+                                // We cannot log the process details here. The output is too large.
+                                process.ThrowIfErrored<MonitorException>(ProcessProxy.DefaultSuccessCodes, errorReason: ErrorReason.MonitorFailed);
+
+                                if (process.StandardOutput.Length > 0)
+                                {
+                                    AmdSmiQueryGpuParser parser = new AmdSmiQueryGpuParser(process.StandardOutput.ToString());
+                                    IList<Metric> metrics = parser.Parse();
+
+                                    if (metrics?.Any() == true)
+                                    {
+                                        this.Logger.LogPerformanceCounters("amd", metrics, startTime, endTime, telemetryContext);

```

Comment:
I know we used LogPerformanceCounters before, but we removed the distinction between counters and metrics. Please use the logger.LogMetrics like in Action classes